src/backend/base/langflow/components/embeddings/text_embedder.py

```diff
@@ -27,26 +36,50 @@ class TextEmbedderComponent(Component):
             info="The message to generate embeddings for.",
         ),
     ]
-
     outputs = [
         Output(display_name="Embedding Data", name="embeddings", method="generate_embeddings"),
     ]
 
     def generate_embeddings(self) -> Data:
-        embedding_model: Embeddings = self.embedding_model
-        message: Message = self.message
+        try:
+            embedding_model: Embeddings = self.embedding_model
+            message: Message = self.message
+
+            # Validate embedding model
+            if not embedding_model:
+                raise ValueError(ERRORS["NO_MODEL"])
+
+            # Extract the text content from the message
+            text_content = message.text if message and message.text else ""
+            if not text_content:
+                raise ValueError(ERRORS["NO_TEXT"])
+
+            # Check if the embedding model has the required attributes
+            if not hasattr(embedding_model, "client") or not embedding_model.client:
+                raise ValueError(ERRORS["NO_CLIENT"])
 
-        # Extract the text content from the message
-        text_content = message.text
+            # Ensure the base URL has proper protocol
+            if hasattr(embedding_model.client, "base_url"):
+                base_url = embedding_model.client.base_url
+                if not base_url.startswith(("http://", "https://")):
+                    embedding_model.client.base_url = f"https://{base_url}"
 
-        # Generate embeddings using the provided embedding model
-        embeddings = embedding_model.embed_documents([text_content])
+            # Generate embeddings using the provided embedding model
+            embeddings = embedding_model.embed_documents([text_content])
 
-        # Assuming the embedding model returns a list of embeddings, we take the first one
-        embedding_vector = embeddings[0] if embeddings else []
+            # Validate embeddings output
+            if not embeddings or not isinstance(embeddings, list):
+                raise ValueError(ERRORS["INVALID_EMBEDDINGS"])
 
-        # Create a Data object to encapsulate the results
-        result_data = Data(data={"text": text_content, "embeddings": embedding_vector})
+            embedding_vector = embeddings[0]
 
-        self.status = {"text": text_content, "embeddings": embedding_vector}
-        return result_data
+        except Exception as e:
+            logging.exception("Error generating embeddings")
+            error_data = Data(data={"text": "", "embeddings": [], "error": str(e)})
+            self.status = {"error": str(e)}
+            return error_data
+        else:
```

Comment:
NIT: the `else` can probably be removed here since the exception returns.

Reply:
you are right!